packages/dmn-editor/tests/e2e/drgRequirements/invalidInformationRequirement.spec.ts

```diff
@@ -0,0 +1,278 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { test, expect } from "../__fixtures__/base";
+import { DefaultNodeName, NodePosition, NodeType } from "../__fixtures__/nodes";
+import { EdgeType } from "../__fixtures__/edges";
+
+test.beforeEach(async ({ editor }) => {
+  await editor.open();
+});
+
+test.describe("Invalid edge - Information Requirement", () => {
+  test.describe("Input Data", () => {
+    test.beforeEach(async ({ palette }) => {
+      await palette.dragNewNode({
+        type: NodeType.INPUT_DATA,
+        targetPosition: { x: 100, y: 100 },
+        thenRenameTo: "Source Node",
+      });
+    });
+
+    test("shouldn't add an Information Requirement edge from Input Data node to Input Data node", async ({
+      palette,
+      nodes,
+      edges,
+    }) => {
+      await palette.dragNewNode({
+        type: NodeType.INPUT_DATA,
+        targetPosition: { x: 300, y: 100 },
+      });
+
+      await nodes.dragNewConnectedEdge({
+        type: EdgeType.INFORMATION_REQUIREMENT,
+        from: "Source Node",
+        to: DefaultNodeName.INPUT_DATA,
+      });
+
+      expect(await edges.get({ from: "Source Node", to: DefaultNodeName.INPUT_DATA })).not.toBeAttached();
+    });
+
+    test("shouldn't add an Information Requirement edge from Input Data node to BKM node", async ({
+      palette,
+      edges,
+      nodes,
+    }) => {
+      await palette.dragNewNode({
+        type: NodeType.BKM,
+        targetPosition: { x: 300, y: 100 },
+      });
+
+      await nodes.dragNewConnectedEdge({
+        type: EdgeType.INFORMATION_REQUIREMENT,
+        from: "Source Node",
+        to: DefaultNodeName.BKM,
+      });
+
+      expect(await edges.get({ from: "Source Node", to: DefaultNodeName.BKM })).not.toBeAttached();
+    });
```

Comment:
ok, here is some update after my investigation, similar visual dim effect test would be possible, it would look like
```
    test.only("should dim the target Input Data node", async ({
      page,
      palette,
      nodes,
    }) => {
      await palette.dragNewNode({
        type: NodeType.BKM,
        targetPosition: { x: 300, y: 100 },
      });

     // lines below could be refactored to single fixture method, e.g. `nodes.startDraggingEdge(from, edgeType)`
      await nodes.select({ name: "Source Node", position: NodePosition.TOP });
      await nodes.get({ name: "Source Node" }).getByTitle("Add Information Requirement edge").hover();
      await page.mouse.down();

      await expect(nodes.get({ name: DefaultNodeName.BKM })).toHaveClass(/.*dimmed/);
    });
```

however, was not able to get this working without actual `class` check, below I put my attempts, that didn't work:
- `await expect(nodes.get({ name: DefaultNodeName.BKM })).toBeDisabled()`
- `await expect(nodes.get({ name: DefaultNodeName.BKM })).not.toBeEditable()`

one more problem is the `dim` implementation, for some nodes we add `dimmed` class to existing `div`:
![image](https://github.com/ljmotta/kie-tools/assets/8044780/4811ba12-9d44-41a7-b7e5-25f37988c08f)

however for other nodes we add new nested `div` with `dimmed` class
![image](https://github.com/ljmotta/kie-tools/assets/8044780/2d575943-3f1d-4992-9ea1-b88c5f9925e6)